src/anyio/_core/_synchronization.py

```diff
@@ -482,6 +509,78 @@ def statistics(self) -> CapacityLimiterStatistics:
         raise NotImplementedError
 
 
+class CapacityLimiterAdapter(CapacityLimiter):
+    _internal_limiter: CapacityLimiter | None = None
+
+    def __new__(cls, total_tokens: int) -> CapacityLimiterAdapter:
+        return object.__new__(cls)
+
+    def __init__(self, total_tokens: int) -> None:
```

Comment:
Line 519 checks if `total_tokens` is not `math.inf` which is a `float`, so should this be reflected in its type?
```suggestion
    def __init__(self, total_tokens: int | float) -> None:
```

Reply:
In that case, the annotation should be `float`, as that already includes integers according to the PEP 484 numeric tower.

Reply:
Well, on the other hand it shouldn't be **any** float. Just `math.inf`.
Shouldn't `int | math.inf` Just Work?

Reply:
No, because `math.inf` is not a type.

Reply:
Neither is `None`, but that works just fine.

Reply:
`None` is explicitly special cased in PEP 484.

Reply:
Ugh. They should just have accepted arbitrary singletons. (I do wonder whether any type checkers do …)

Reply:
I think it should be `int | Literal[math.inf]`, but mypy doesn't like it. Maybe it's a bug in mypy?

Reply:
No, it's not. Floats are not valid literals according to [PEP 586](https://peps.python.org/pep-0586/#illegal-parameters-for-literal-at-type-check-time).

Reply:
Thanks.
BTW, why not using `None` to mean an infinite number of tokens? `math.inf` carries a direct meaning, but `None` also means "I don't want to specify a finite number of tokens", which easily translates to a infinite number of tokens IMO.

Reply:
It couldn't been done, but I was following trio's API design which uses `math.inf` instead.